src/core/Akka.Remote/EndpointManager.cs

```diff
@@ -492,196 +543,248 @@ protected void Accepting(object message)
-                        eventPublisher.NotifyListeners(new QuarantinedEvent(quarantine.RemoteAddress, quarantine.Uid.Value));
-                    }
-                })
-                .With<Send>(send =>
+                        var handle = assoc.Association.AsInstanceOf<AkkaProtocolHandle>();
+                        var drop = matchesQuarantine(handle);
+                        if (drop)
+                            handle.Disassociate();
+                        return !drop;
+                    }).ToList();
+                    return new KeyValuePair<IActorRef, List<InboundAssociation>>(x.Key, associations);
+                }).ToDictionary(k => k.Key, v => v.Value);
+
+                if (quarantine.Uid.HasValue)
                 {
-                    var recipientAddress = send.Recipient.Path.Address;
-                    Func<int?, IActorRef> createAndRegisterWritingEndpoint = refuseUid => endpoints.RegisterWritableEndpoint(recipientAddress,
-                        CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse,
-                            _transportMapping[send.Recipient.LocalAddressToUse], settings, writing: true,
-                            handleOption: null, refuseUid: refuseUid), refuseUid);
-
-                    endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
-                        .With<Pass>(
-                            pass =>
-                            {
-                                pass.Endpoint.Tell(send);
-                            })
-                        .With<Gated>(gated =>
+                    _endpoints.MarkAsQuarantined(quarantine.RemoteAddress, quarantine.Uid.Value, Deadline.Now + _settings.QuarantineDuration);
+                    _eventPublisher.NotifyListeners(new QuarantinedEvent(quarantine.RemoteAddress, quarantine.Uid.Value));
+                }
+            });
+
+            Receive<Send>(send =>
+            {
+                var recipientAddress = send.Recipient.Path.Address;
+                Func<int?, IActorRef> createAndRegisterWritingEndpoint = refuseUid => _endpoints.RegisterWritableEndpoint(recipientAddress,
+                    CreateEndpoint(recipientAddress, send.Recipient.LocalAddressToUse,
+                        _transportMapping[send.Recipient.LocalAddressToUse], _settings, writing: true,
+                        handleOption: null, refuseUid: refuseUid), uid: null, refuseUid: refuseUid);
+
+                _endpoints.WritableEndpointWithPolicyFor(recipientAddress).Match()
+                    .With<Pass>(
+                        pass =>
                         {
-                            if(gated.TimeOfRelease.IsOverdue) createAndRegisterWritingEndpoint(null).Tell(send);
-                            else Context.System.DeadLetters.Tell(send);
+                            pass.Endpoint.Tell(send);
```

Comment:
If a `Send` is written to the `EndpointManager` with a `Pass` policy, forward the message onto the appropriate writer